### ISSUE TITLE
Required variable not set: variable despite the fact that the variables and values are already defined

### ISSUE THREAD

**Issue by jpdoria** [hashicorp/terraform#15894]
### Terraform Version

Terraform v0.10.2

### Terraform Configuration Files

**terraform.tfvars**

```
aws_access_key = "XXXXXX"
aws_secret_key = "XXXXXX"
aws_key_path = "/Users/user/.ssh/XXXXXX.pem"
aws_key_name = "XXXXXX"
aws_iam_role = "arn:aws:iam::XXXXXX:role/XXXXXX"
aws_role_session = "XXXXXX"
aws_region = "us-east-1"
```

**vars.tf**

```
variable "aws_region" {}
variable "aws_access_key" {}
variable "aws_secret_key" {}
variable "aws_key_path" {}
variable "aws_key_name" {}
variable "aws_iam_role" {}
variable "aws_role_session" {}
```

**provider.tf**

```
provider "aws" {
  version    = "~> 0.1"
  access_key = "${var.aws_access_key}"
  secret_key = "${var.aws_secret_key}"
  region     = "${var.aws_region}"

  assume_role = {
    role_arn     = "${var.aws_iam_role}"
    session_name = "${var.aws_role_session}"
  }
}
```

**Project Structure**

```
terraform
├── dev
│   ├── main.tf
│   ├── provider.tf
│   ├── terraform.tfvars
│   ├── terraform_remote_state.tf
│   └── vars.tf
├── modules
│   ├── eb
│   │   ├── ha
│   │   └── single
│   │       ├── main.tf
│   │       ├── outputs.tf
│   │       └── vars.tf
│   ├── r53
│   │   ├── main.tf
│   │   ├── outputs.tf
│   │   └── vars.tf
│   ├── rds
│   │   ├── main.tf
│   │   ├── outputs.tf
│   │   └── vars.tf
│   └── vpc
│       ├── main.tf
│       ├── outputs.tf
│       └── vars.tf
├── prod 
├── qa
```

### Expected Behavior
Terraform should be able to read the variables

### Actual Behavior

```
# terraform validate
There are warnings and/or errors related to your configuration. Please
fix these before continuing.

Errors:

  * 7 error(s) occurred:

* Required variable not set: aws_access_key
* Required variable not set: aws_secret_key
* Required variable not set: aws_key_path
* Required variable not set: aws_key_name
* Required variable not set: aws_iam_role
* Required variable not set: aws_role_session
* Required variable not set: aws_region
#
```

### Steps to Reproduce

`terraform validate`

### Workaround

**vars.tf**

```
variable "aws_region" {
  default = "foo"
}

variable "aws_access_key" {
  default = "foo"
}

variable "aws_secret_key" {
  default = "foo"
}

variable "aws_key_path" {
  default = "foo"
}

variable "aws_key_name" {
  default = "foo"
}

variable "aws_iam_role" {
  default = "foo"
}

variable "aws_role_session" {
  default = "foo"
}
```

**Comment by apparentlymart** [comment]
Hi @jpdoria! Sorry this isn't working as expected.

I think this might be the same issue as #15733. If so, the workaround until that's fixed would be to explicitly specify the vars file:

```
$ terraform validate -var-file=terraform.tfvars
```

**Comment by luckytaxi** [comment]
I'm having the same issues with v0.10.7

**Comment by dev-drprasad** [comment]
I am also having the same issues with v0.10.7

**Comment by apparentlymart** [comment]
Hi all,

We added a fix for this in #15938 which was confirmed as fixed in v0.10.5 per discussion on #15733. It looks like some of you have a problem with similar symptoms, but the cause must be different if it's still present after v0.10.5.

In order to understand what's different about your situation, could one of you please open a new top-level issue with the details requested by the issue template? That way we will hopefully have enough information to reproduce the issue, whereas the details given in the description of this one no longer seem to reproduce.

Thanks!
